src/Symfony/Component/Yaml/Parser.php

```diff
@@ -221,7 +221,14 @@ private function doParse($value, $flags)
                 try {
                     Inline::$parsedLineNumber = $this->getRealCurrentLineNb();
                     $i = 0;
-                    $key = Inline::parseScalar($values['key'], 0, null, $i, !(Yaml::PARSE_KEYS_AS_STRINGS & $flags));
+                    $evaluateKey = !(Yaml::PARSE_KEYS_AS_STRINGS & $flags);
```

Comment:
https://github.com/symfony/symfony/pull/21774 introduced a small bc break anyway and this only allows people to go back to the old behavior by using this flag, right?